packages/botonic-core/index.d.ts

```diff
@@ -37,37 +46,61 @@ export interface Route {
   text?: StringMatcher
   type?: StringMatcher
   intent?: StringMatcher
+  input?: (input: Input) => boolean
   session?: (session: Session) => boolean
   action: React.ReactNode
 }
 
+type Routes = Route[] | ((_: { input: Input; session: Session }) => Route[])
+
 // Desk
 export declare function humanHandOff(
   session: Session,
   queue_name: string,
   on_finish: { payload?: any; path?: any }
 ): Promise<void>
+
 export declare function getOpenQueues(
   session: Session
 ): Promise<{ queues: string[] }>
+
 export declare function storeCaseRating(
   session: Session,
   rating: number
 ): Promise<any>
 
-type Routes = Route[] | ((_: { input: Input; session: Session }) => Route[])
+export interface Plugin {
+  pre(_: { input: Input; session: Session; lastRoutePath: string })
+  post(_: {
+    input: Input
+    session: Session
+    lastRoutePath: string
+    response: string
```

Comment:
with what is associated this response param?

Reply:
good catch! see d37df80 for a fix and description of the semantics